packages/core/core-flows/src/draft-order/workflows/update-draft-order-shipping-method.ts

```diff
@@ -0,0 +1,154 @@
+import { draftOrderFieldsForRefreshSteps } from "../utils/fields"
+import { refreshDraftOrderAdjustmentsWorkflow } from "./refresh-draft-order-adjustments"
+
+export const updateDraftOrderShippingMethodWorkflowId =
+  "update-draft-order-shipping-method"
+
+export interface UpdateDraftOrderShippingMethodWorkflowInput {
+  /**
+   * The ID of the order to update the shipping method in its edit.
+   */
+  order_id: string
+  data: {
+    /**
+     * The ID of the shipping method to update.
+     */
+    shipping_method_id: string
+    /**
+     * The ID of the shipping option to associate with the shipping method.
+     */
+    shipping_option_id?: string
+    /**
+     * Set a custom amount for the shipping method.
+     */
+    custom_amount?: BigNumberInput
+    /**
+     * A note viewed by admins only related to the shipping method.
+     */
+    internal_note?: string | null
+  }
+}
+
+export const updateDraftOrderShippingMethodWorkflow = createWorkflow(
+  updateDraftOrderShippingMethodWorkflowId,
+  function (input: WorkflowData<UpdateDraftOrderShippingMethodWorkflowInput>) {
+    const order: OrderDTO = useRemoteQueryStep({
+      entry_point: "orders",
+      fields: ["id", "status", "is_draft_order"],
+      variables: { id: input.order_id },
+      list: false,
+      throw_if_key_not_found: true,
+    }).config({ name: "order-query" })
+
+    const orderChange: OrderChangeDTO = useRemoteQueryStep({
+      entry_point: "order_change",
+      fields: ["id", "status", "version", "actions.*"],
+      variables: {
+        filters: {
+          order_id: input.order_id,
+          status: [OrderChangeStatus.PENDING, OrderChangeStatus.REQUESTED],
+        },
+      },
+      list: false,
+    }).config({ name: "order-change-query" })
+
+    validateDraftOrderChangeStep({ order, orderChange })
+
+    const { before, after } = updateDraftOrderShippingMethodStep({
+      order_id: input.order_id,
+      shipping_method_id: input.data.shipping_method_id,
+      shipping_option_id: input.data.shipping_option_id,
+      amount: input.data.custom_amount,
+    })
+
+    updateOrderTaxLinesWorkflow.runAsStep({
+      input: {
+        order_id: order.id,
+        shipping_method_ids: [input.data.shipping_method_id],
+      },
+    })
+
+    const refetchedOrder = useRemoteQueryStep({
+      entry_point: "orders",
+      fields: draftOrderFieldsForRefreshSteps,
+      variables: { id: input.order_id },
+      list: false,
+      throw_if_key_not_found: true,
+    }).config({ name: "refetched-order-query" })
+
+    const appliedPromoCodes = transform(refetchedOrder, (refetchedOrder) => {
+      const promotionLink = (refetchedOrder as any).promotion_link
+
+      if (!promotionLink) {
+        return []
+      }
+
+      if (Array.isArray(promotionLink)) {
+        return promotionLink.map((promo) => promo.promotion.code)
+      }
+
+      return [promotionLink.promotion.code]
```

Comment:
question: Why do we need to query the link here? Nothing wrong with it, just seems like an anti pattern